shared/src/main/scala/org/atnos/eff/FutureEffect.scala

```diff
@@ -0,0 +1,226 @@
+        val newCallback = { (sexs: ScheduledExecutorService, ec: ExecutionContext) =>
+          val ffRan = ff.runNow(sexs, ec)
+          val faRan = fa.runNow(sexs, ec)
+          faRan.flatMap(a => ffRan.map(f => f(a))(ec))(ec)
+        }
+        TimedFuture(newCallback)
+      }
+      override def toString = "Applicative[TimedFuture]"
+    }
+
+    implicit final def MonadTimedFuture: Monad[TimedFuture] = new Monad[TimedFuture] {
+      override def pure[A](x: A) = TimedFuture((_, _) => Future.successful(x))
+      override def flatMap[A, B](fa: TimedFuture[A])(f: (A) => TimedFuture[B]): TimedFuture[B] =
+        TimedFuture[B]((sexs, ec) => fa.runNow(sexs, ec).flatMap(f(_).runNow(sexs, ec))(ec))
+      override def tailRecM[A, B](a: A)(f: (A) => TimedFuture[Either[A, B]]): TimedFuture[B] =
+        TimedFuture[B]({ (sexs, ec) =>
+          def loop(va: A): Future[B] = f(va).runNow(sexs, ec).flatMap {
+            case Left(na) => loop(na)
+            case Right(nb) => Future.successful(nb)
+          }(ec)
+          loop(a)
+        })
+      override def toString = "Monad[TimedFuture]"
+    }
+  }
+
+trait FutureTypes {
+
+  type _future[R] = TimedFuture |= R
+  type _Future[R] = TimedFuture <= R
+}
+
+trait FutureCreation extends FutureTypes {
+
+  final def futureWithExecutors[R: _future, A](c: (ScheduledExecutorService, ExecutionContext) => Future[A], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture(c, timeout))
+
+  final def future[R: _future, A](c: => Future[A], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture((_, _) => c, timeout))
+
+  final def futureCB[R: _future, A](c: (ScheduledExecutorService, ExecutionContext) => Future[A],
+                                    timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture(c, timeout))
+
+  final def futureFailed[R: _future, A](t: Throwable): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture((_, _) => Future.failed(t)))
+
+  final def futureFromEither[R: _future, A](e: Throwable Either A): Eff[R, A] =
+    e.fold(futureFailed[R, A], Eff.pure[R, A])
+
+  final def futureDelay[R: _future, A](a: => A, timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture((_, ec) => Future(a)(ec), timeout))
+
+  final def futureFork[R: _future, A](a: => A, ec: ExecutionContext, timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture((_, _) => Future(a)(ec), timeout))
+
+  final def futureDefer[R: _future, A](a: => Future[A], timeout: Option[FiniteDuration] = None): Eff[R, A] =
+    send[TimedFuture, R, A](TimedFuture((_, _) => a, timeout))
+
+}
+
+trait FutureInterpretation extends FutureTypes {
+
+  final def futureAttempt[R, A](e: Eff[R, A])(implicit future: TimedFuture /= R): Eff[R, Throwable Either A] = {
```

Comment:
It would be nice to add an interpreter to get back a `Future[A]`
```scala
def runFuture[A](e: Eff[Fx1[TimedFuture], A])(implicit ec: ExecutionContext, s: SchedulerService): Future[A]
```